core/base/inc/TString.h

```diff
@@ -132,15 +133,19 @@ friend TString operator+(const TString &s1, const TString &s2);
 friend TString operator+(const TString &s,  const char *cs);
 friend TString operator+(const char *cs, const TString &s);
 friend TString operator+(const TString &s, char c);
-friend TString operator+(const TString &s, Long_t i);
-friend TString operator+(const TString &s, ULong_t i);
-friend TString operator+(const TString &s, Long64_t i);
-friend TString operator+(const TString &s, ULong64_t i);
+template<class T, class = typename std::is_signed<T>::type>
+friend TString operator+(const TString &s, T i);
```

Comment:
You only need *two* declarations (`is_integral` and `is_floating_point`); they cover all `i`, `u` etc!

And this needs the definition right here, or else you'll get the errors we see.

Reply:
But, using only`is_integral` as one function would lose the one bit for the `unsigned` vs `signed` inputs because depending on the choice of cast, it would remove the sign of `signed` for more precision or it would remove more precision of an `unsigned` and return the wrong string.

Reply:
Indeed - so maybe have a floating point, a signed and an unsigned specialization?